pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go

```diff
@@ -441,29 +442,22 @@ func checkBond(b *Bond) error {
 	return result.ErrorOrNil()
 }
 
+//nolint: gocyclo
 func checkWireguard(b *DeviceWireguardConfig) error {
 	var result *multierror.Error
 
-	// avoid pulling in wgctrl code to keep machinery dependencies slim
-	checkKey := func(key string) error {
-		raw, err := base64.StdEncoding.DecodeString(key)
-		if err != nil {
-			return err
-		}
-
-		if len(raw) != 32 {
-			return fmt.Errorf("wrong key %q length: %d", key, len(raw))
-		}
-
-		return nil
+	if b.WireguardPrivateKey == "" && !b.WireguardEnableAutomaticNodes {
```

Comment:
if I understand it correctly, private key should be always available? even with kubespan?

Reply:
Yes, private key must be available.   Ideally (and originally) we could generate the keypair as needed, but since we both wanted to avoid updating the machine config on the host _and_ we don't have a place yet to avoid secrets, we are left with generating this supply-side.

Reply:
Each node now indigenously generated wg private keys as needed, if they are not supplied (and they are _not_ now generated by the config generator).  They are stored in the state directory.

This does have the side-effect that during the installer, a temporary one is generated and lost on reboot, which ends up with duplicate peers with differing keys for a while.  That will need to be fixed ultimately, but the new expiration and most-recent-route routines allay the impact of this complication.

Reply:
now generates private key at runtime, storing it to a file on state... but if that gets wiped, it's not a big deal.  It will be regenerated and redistributed.  There will just be a little time delay (a few minutes) for full propagation in such a case.